elements/elix-mixins/src/symbols.js

```diff
@@ -0,0 +1,113 @@
+import createSymbol from './createSymbol';
+
+/**
+ * A collection of Symbol objects for standard component properties and methods.
+ *
+ * These Symbol objects are used to allow mixins and a component to internally
+ * communicate, without exposing these properties and methods in the component's
+ * public API.
+ *
+ * To use these Symbol objects in your own component, include this module and
+ * then create a property or method whose key is the desired Symbol.
+ *
+ *     import 'SingleSelectionMixin' from 'elix-mixins/src/SingleSelectionMixin';
+ *     import 'symbols' from 'elix-mixins/src/symbols';
+ *
+ *     class MyElement extends SingleSelectionMixin(HTMLElement) {
+ *       [symbols.itemSelected](item, selected) {
+ *         // This will be invoked whenever an item is selected/deselected.
+ *       }
+ *     }
+ *
+ * @module symbols
+ */
+const symbols = {
+
+  /**
+   * Symbol for the `defaults` property.
+   *
+   * This property can be used to set or override defaults that will be applied
+   * to a new component instance. When implementing this property, take care to
+   * first acquire any defaults defined by the superclass. The standard idiom is
+   * as follows:
+   *
+   *     get [symbols.defaults]() {
+   *       const defaults = super[symbols.defaults] || {};
+   *       // Set or override default values here
+   *       defaults.customProperty = false;
+   *       return defaults;
+   *     }
+   *
+   * @var {object} defaults
+   */
+  defaults: createSymbol('defaults'),
+
+  /**
+   * Symbol for the `raiseChangeEvents` property.
+   *
+   * This property is used by mixins to determine whether they should raise
+   * property change events. The standard HTML pattern is to only raise such
+   * events in response to direct user interactions. This property can be used
+   * to manage events as follows.
+   *
+   * First, UI event listeners should set this property to `true` at the start
+   * of the event handler, then `false` at the end:
+   *
+   *     this.addEventListener('click', event => {
+   *       this[symbols.raiseChangeEvents] = true;
```

Comment:
It would be possible. I'm worried that saving the line of code ends up costing us some conceptual overhead.

If we introduce a helper method to set the flag, then we inevitably end up with two things: 1) a method to set the property and then clear it, and 2) the property itself storing the flag:

    class MyElement extends HTMLElement {

      constructor() {
        this.addEventListener('some-event', event => {
          this[symbols.raiseChangeEventsMethod](() => {
            this.foo = 'hello';
          });
        })
      }

      get foo() {
        return this._foo;
      }
      set foo(value) {
        this._foo = value;
        if (this[symbols.raiseChangeEventsProperty]) {
          this.dispatchEvent(new CustomEvent('foo-changed'));
        }
      }
      
    }

Note that the constructor calls `raiseChangeEventsMethod` and the property setter calls `raiseChangeEventsProperty`. No matter what we name that method and property, there are two things to look at. Their relationship can be suggested by the way we name them, but ultimately, the developer has no way of knowing how they're actually connected.

If we make the developer set the flag directly, they need to write an additional line of code in the event handler:

    class MyElement extends HTMLElement {

      constructor() {
        this.addEventListener('some-event', event => {
          this[symbols.raiseChangeEvents] = true;
          this.foo = 'hello';
          this[symbols.raiseChangeEvents] = false;
        })
      }

      get foo() {
        return this._foo;
      }
      set foo(value) {
        this._foo = value;
        if (this[symbols.raiseChangeEvents]) {
          this.dispatchEvent(new CustomEvent('foo-changed'));
        }
      }
      
    }

But I think this approach is clearer to the uninitiated. There's no magic going on, just a flag being set and cleared. That feels, perhaps, like a blunt solution, but its simplicity has merit. As long as I can read ES6, I can read the above and predict what will happen without further explanation.

As soon as we introduce a method, we also introduce some degree of magic. That puts us just a little further down the road to becoming a framework whose abstractions might be useful, but will be unfamiliar to the uninitiated. People may also push further down that path. Someone might look at the `addEventListener` pattern above and notice that all the event handlers have to make the `raiseChangeEventsMethod` call. They might reasonably ask, "Why not bake the flag-setting helper into the event listener call"?

      constructor() {
        this.addEventListenerAndRaiseChangeEvents('some-event', event => {
          this.foo = 'hello';
        })
      }

That code is even shorter, and some people might reasonably prefer it. A project that prefers that could write their own helper to do that. But for the project's own mixins/components, I think we're better off with plainer code.

It's a matter of personal taste, but my own preference is to stay as close to the metal as possible here. I'm open to other perspectives, though.